packages/core/src/lib/light_push/index.ts

```diff
@@ -38,10 +35,20 @@ type PreparePushMessageResult =
       error: SendError;
     };
 
+type CoreSendResult =
```

Comment:
nit: I like this pattern but it if we use it elsewhere we may want to use a library like fp-ts which adds an `Either` type: https://gcanti.github.io/fp-ts/modules/Either.ts.html

Reply:
nice. would be interesting to tackle with #1694

Reply:
upon digging through the API more, i'd like to discard this for #1694 strictly because I think its addition will be maximised if we switch a lot of code to `fp-ts`'s functional way which does make things unreadable and introduces a learning curve.
im happy to discuss more perhaps in the PM call cc @weboko

Reply:
I set up a generic type I am talking about in the above comment here: https://github.com/waku-org/js-waku/commit/b308d71cd418cc5c47098f8d0da72bdc3f3ac2f0

Reply:
I'm realizing `fp-ts` is probably too opinionated for any types that will reach the library consumer. I just worry if we roll our own generic, we will, over time, replicate a lot of the same code. Something to consider for the future, but for now I think this type works well.